src/rebar_dialyzer.erl

```diff
@@ -0,0 +1,252 @@
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil -*-
+%% ex: ts=4 sw=4 et
+%% -------------------------------------------------------------------
+%%
+%% rebar: Erlang Build Tools
+%%
+%% Copyright (c) 2015 Tuncer Ayaz
+%%
+%% Permission is hereby granted, free of charge, to any person obtaining a copy
+%% of this software and associated documentation files (the "Software"), to deal
+%% in the Software without restriction, including without limitation the rights
+%% to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+%% copies of the Software, and to permit persons to whom the Software is
+%% furnished to do so, subject to the following conditions:
+%%
+%% The above copyright notice and this permission notice shall be included in
+%% all copies or substantial portions of the Software.
+%%
+%% THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+%% IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+%% FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+%% AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+%% LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+%% OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+%% THE SOFTWARE.
+%% -------------------------------------------------------------------
+-module(rebar_dialyzer).
+
+-export([
+         dialyze/2,
+         'build-plt'/2,
+         'check-plt'/2,
+         'delete-plt'/2
+        ]).
+
+%% for internal use only
+-export([info/2]).
+
+-include("rebar.hrl").
+
+%% ===================================================================
+%% Public API
+%% ===================================================================
+
+%% TODO:
+%%
+%%  * How can we configure the filtering of warnings without exact line
+%%    numbers as a way to avoid updating the config whenever the file
+%%    in question is modified (and a warning's line number changes)?
+%%    Maybe use '_' for line number in dialyzer_ignored_warnings or a
+%%    separate dialyzer_ignored_warnings_noline (name to be determined)
+%%    list, and subtract that from the dialyzer:run/2 result.
+
+dialyze(Config, AppFile) ->
+    {NewConfig, Plt} = plt(Config, AppFile),
+    ok = check_plt_existence(Plt),
+
+    Opts = [
+            {analysis_type, succ_typings},
+            %% http://erlang.org/pipermail/erlang-bugs/2015-February/004781.html
+            %% TODO: remove once the minimum required Erlang/OTP release includes
+            %% a Dialyzer version without the bug, or alternatively add a config
+            %% option to always check the PLT, as this may be needed by some user.
+            {check_plt, false},
```

Comment:
Now that `{check_plt, false}` is used the PLT will not be check for consistency when running `dialyzer`, should it be checked? With `{check_plt, true}` any warnings generated when checking the PLT are ignored. To copy this behaviour there would be `plt_check` run before `succ_typings`, but ignoring any warnings from `plt_check`.